### ISSUE TITLE
salt_function_status() counter doesn't increment

### ISSUE THREAD

**Issue by rjel2159** [kpetremann/salt-exporter#26]
The salt_function_status() counter doesn't increment but always stays 1 - here's an example when running total of 7 states on a single minion:
```
#  curl -s http://localhost:2112/metrics|grep ^salt_
salt_function_responses_total{function="state.apply",state="highstate",success="true"} 3
salt_function_responses_total{function="state.highstate",state="highstate",success="true"} 2
salt_function_responses_total{function="state.sls",state="sys.passwd",success="true"} 2
salt_function_status{function="state.apply",minion="gen12472",state="highstate"} 1
salt_function_status{function="state.highstate",minion="gen12472",state="highstate"} 1
salt_function_status{function="state.sls",minion="gen12472",state="sys.passwd"} 1
salt_responses_total{minion="gen12472",success="true"} 7
#

**Comment by rjel2159** [comment]
> Hello @rjel2159,
> 
> `salt_function_status` is a gauge, not a counter. 1 = ok 0 = not ok
> 
> See the help of the metric:
> 
> ```
> $ curl -s http://localhost:2112/metrics | grep  salt_function_status
> # HELP salt_function_status Last function/state success, 0=Failed, 1=Success
> ```

I see, thanks for your response. It'd then be useful to have a metric such as salt_function_responses(minion=<minion>, function=<function>, state=<state>) that would record the count.

**Comment by kpetremann** [comment]
The cardinality with `salt_function_status` is already high (depending on your filter). It would double.

Why do you need that much details?

**Comment by kpetremann** [comment]
:+1: 
If your need was only to troubleshoot an issue, I would have suggested either to use the salt-live tool in this repo, or to use dedicated returner to sent some metrics to a Prometheus Gateway.

Let me think about what would be the best approach.
I would like to avoid current user to be impacted by such changes.

Something like a "verbose-metric" flag or something like this, which would either:
- create a new metric with more details
- or, most likely, add the minion label to `salt_function_responses_total` metric

I also need to think about the other current/future metrics. Users might expect to have the same option for these. 
So I might need to rethink the configuration to better scale the different options, while keeping this exporter simple to use and maintain.

**Comment by rjel2159** [comment]
Awesome, thanks so much for the update!


     On Thursday, June 29, 2023 at 12:25:05 PM EDT, Kevin Petremann ***@***.***> wrote:  
 
 


Hi @rjel2159,

FYI the work is mostly done, I need to add more tests and update the documentation, as I completely refacto the configuration of the exporter.

You can already test it using the configurable_metrics branch.

You will need a configuration file called config.yml containing at least:
metrics:  salt_function_responses_total:    enabled: true    add-minion-label: true  salt_scheduled_job_return_total:    enabled: true    add-minion-label: true

**Comment by kpetremann** [comment]
The pre-release is available for testing: https://github.com/kpetremann/salt-exporter/releases/tag/v0.7.0-pre.1

The matching documentation is here: https://github.com/kpetremann/salt-exporter/blob/configurable_metrics/README.md

**Comment by kpetremann** [comment]
The feature is now available in the v0.7.0 which has just been released.

@rjel2159 feel free to reopen this issue, or another one, if you encounter any problem.
